git-version/src/__tests__/getGitVersion.test.ts

```diff
@@ -0,0 +1,86 @@
+import { SemVer } from 'semver'
+import { when } from 'jest-when'
+
+import {
+  Commit,
+  Repository,
+  Tag,
+  getInput,
+  getRepository,
+  getCommit,
+  getTagList,
+} from '../github'
+
+import {
+  ParsedAncestorTag,
+  findAncestorTag,
+  buildVersionString,
+} from '../versioning'
+
+import { getGitVersion } from '../getGitVersion'
+
+jest.mock('../github')
+jest.mock('../versioning')
+
+const REPOSITORY: Repository = {
+  owner: 'owner-name',
+  repo: 'repo-name',
+}
+
+const COMMIT: Commit = {
+  commit: 'abc123',
+}
+
+const TAG_LIST: Tag[] = [
+  { tag: 'v1.2.3' },
+  { tag: 'v4.5.6' },
+  { tag: 'v7.8.9' },
+]
+
+const ANCESTOR_TAG: ParsedAncestorTag = {
+  tag: 'v4.5.6',
+  distance: 42,
+  semver: new SemVer('v4.5.6'),
+}
+
+describe('get git version', () => {
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('should get the list of tags, find the closest ancestor, and increment it', async () => {
+    when(getInput).calledWith('tag-prefix').mockReturnValue('v')
+    when(getInput).calledWith('bump').mockReturnValue('preminor')
+    when(getInput).calledWith('prerelease-id').mockReturnValue('alpha')
+    when(getInput).calledWith('snapshot-id').mockReturnValue('dev')
+    when(getCommit).calledWith().mockReturnValue(COMMIT)
+    when(getRepository).calledWith().mockReturnValue(REPOSITORY)
+
+    when(getTagList)
+      .calledWith({ tagPrefix: 'v', repository: REPOSITORY })
+      .mockResolvedValue(TAG_LIST)
+
+    when(findAncestorTag)
+      .calledWith({
+        tagPrefix: 'v',
+        tags: TAG_LIST,
+        commit: COMMIT,
+        repository: REPOSITORY,
+      })
+      .mockResolvedValue(ANCESTOR_TAG)
+
+    when(buildVersionString)
+      .calledWith({
+        ancestor: ANCESTOR_TAG,
+        commit: COMMIT,
+        bump: 'preminor',
+        prereleaseId: 'alpha',
+        snapshotId: 'dev',
+      })
+      .mockReturnValue('v0.0.0')
+
+    const version = await getGitVersion()
```

Comment:
i know you're awaiting here so the js runtime will pause execution in this test block until, but i think it would be a bit safer for async tests to assert on the number of expectations we'll have
```js
expect.assertions(1);
```

Reply:
Happy to add this in (or nuke the `await` syntax, which I don't like but used here because it made the test cleaner to read), but I ended up getting pretty influenced by [ava's thoughts on the subject](https://github.com/avajs/ava/blob/main/docs/recipes/when-to-use-plan.md) of when to declare how many expectations you're making.

(After trying it on a few projects, I wouldn't choose ava, but I think their take on this subject is solid)